Library/Homebrew/extend/fileutils.rb

```diff
@@ -10,6 +11,19 @@ def mktemp(prefix = name)
     prev = pwd
     tmp  = Dir.mktmpdir(prefix, HOMEBREW_TEMP)
 
+    # Make sure files inside the temporary directory have the same group as the
+    # brew instance.
+    #
+    # Reference from `man 2 open`
+    # > When a new file is created, it is given the group of the directory which
+    # contains it.
+    if (File.grpowned? HOMEBREW_BREW_FILE)
+      chown(nil, File.stat(HOMEBREW_BREW_FILE).gid, tmp)
+    else
+       #chown(nil, Etc.getpwnam(Etc.getlogin).gid, tmp)
+       chown(nil, Process.gid, tmp)
```

Comment:
Fair enough. Is there any reason the tests bots cannot have brew-instance-group-brew-user-group agreement?

Also, which fallback do you prefer?
a) ``chown(nil, Etc.getpwnam(Etc.getlogin).gid, tmp)`` thereby setting group to staff
b) ``chown(nil, Process.gid, tmp)`` thereby setting group to staff
c) do nothing when ```if (File.grpowned? HOMEBREW_BREW_FILE)``` fails, and let wheel get carried over
d) something else

Also, even with a fallback in place, do we also want to catch ```EPERM``` just in case?